app/models/offer.rb

```diff
@@ -0,0 +1,14 @@
+class Offer < ApplicationRecord
+  belongs_to :employee
+  belongs_to :selection_process
+
+  belongs_to :message
+
+  delegate :candidate, to: :selection_process
+  delegate :position, to: :selection_process
+
+  enum hiring_scheme: { clt: 0, contractor: 5, internship: 10 }
+  enum status: { pending: 0, accepted: 5, canceled: 10 }
+
+  validates :salary, :hiring_scheme, :start_date, :message, presence: true
+end
```

Comment:
Como esse model basicamente 'herda' os campos de position, já que a proposta pode ser modificada conforme candidato e funcionário conversam, talvez seria interessante utilizar uma Single Table Inheritance (STI) para tentar manter o projeto DRY (don't repeat yourself).

Source: https://eewang.github.io/blog/2013/03/12/how-and-when-to-use-single-table-inheritance-in-rails/